### PR TITLE
mgr/dashboard: iops optimized osd creation

### PR DESCRIPTION
**Pull request by pereman2**
As a first iteration in this option we will enable the iops optimzed option with non rotational devices due to the fact that we can't infer if a devices is an nvme through the current inventory api. This should/might change in the future.

Signed-off-by: Pere Diaz Bou <pdiazbou@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by nizamial09 — APPROVED** [review]

**Review by avanthakkar — APPROVED** [review]

File: src/pybind/mgr/dashboard/controllers/osd.py [hunk 1]
```diff
@@ -489,24 +499,25 @@ def deployment_options(self):
...
             for device in inventory_host.devices.devices:
                 if device.available:
-                    devices[device.path] = device
                     if device.human_readable_type == 'hdd':
                         hdds += 1
-                    elif device.human_readable_type == 'ssd':
+                    else:
                         ssds += 1
-                    elif device.human_readable_type == 'nvme':
+                        # we still don't know how to infer nvmes
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/controllers/osd.py:510]: What about creating a tracker issue and even pasting here the link to that (`# TODO:...`).

Comment (pereman2) [reply src/pybind/mgr/dashboard/controllers/osd.py:510]: @epuertat done

**Comment by epuertat** [comment]
You need to rebase the target branch, since the following error was fixed in master a few weeks ago:
```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
apache-libcloud 3.5.1 requires requests>=2.26.0, but you have requests 2.25.1 which is incompatible.
```

**Comment by pereman2** [comment]
jenkins test dashboard

**Comment by pereman2** [comment]
jenkins test cephadm dashboard

**Comment by pereman2** [comment]
jenkins test dashboard cephadm

File: src/pybind/mgr/dashboard/controllers/osd.py [hunk 2]
```diff
@@ -493,24 +503,26 @@ def deployment_options(self):
...
+                        # Tracker: https://tracker.ceph.com/issues/55728
                         nvmes += 1
+
         if hdds:
             res.options[OsdDeploymentOptions.COST_CAPACITY].available = True
             res.recommended_option = OsdDeploymentOptions.COST_CAPACITY
         if ssds:
             res.options[OsdDeploymentOptions.THROUGHPUT].available = True
             res.recommended_option = OsdDeploymentOptions.THROUGHPUT
+        if nvmes:
```
Comment (sunilangadi2) [inline src/pybind/mgr/dashboard/controllers/osd.py:524]: @pereman2 all **hdds** , **ssds** and **nvmes** are holding number of respective devices it has, But based on that how is it deciding which osd deployment option to recommend 
Suppose consider cluster has 3 osd nodes in that one host has **ssds** and other two are **nvme** from the code read which osd deployment option it recommend?
Can you please explain the logic on how osd deployment option is recommended?

**Pull request merged**